enarxbot-pr-merge

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+import bot
+
+import json
+import sys
+import os
+
+MERGEABLE_QUERY = '''
+query($owner:String!, $repo:String!, $cursor:String) {
```

Comment:
What is a `$cursor`?

Reply:
@lkatalin See my response to your top-level question for more background on GraphQL.

Since we're calling GraphQL directly now, a big thing we need to deal with is **API pagination**. Asking for all issues in a repository, for example, may be feasible for a small repo, but for a large one it may constitute hundreds of thousands of things. Returning all of them in one JSON object just isn't feasible in some cases.

The solution to this (from an API perspective) is to paginate these results. So, for example, if I ask for issues from a project, Github will return only the first 100. If I want the next 100, I need to "turn the page" and ask for them in a separate request.

This is necessary for Github, but it's quite annoying for us as consumers of the API that just want a clean data structure of everything we ask for. Our solution is to 

1. request pages of objects until there are no more pages
2. stitch these pages together locally to return a unified, usable set for consumers

Basically, whenever you ask Github for paginated data, and properly specify a few arguments (of which `$cursor` is one), our `graphql()` function will automatically "depaginate" the data.

You can see the code that does this here, along with instructions for how to depaginate: https://github.com/enarx/bot/blob/master/bot.py#L72